src/LinQuadOptInterface.jl

```diff
@@ -110,55 +110,52 @@ This function updates all of the references in `m`
 after we have deleted row `row` in the affine constraint matrix.
 """
 function shift_references_after_delete_affine!(m, row)
-    for scalar_affine in [
-            cmap(m).less_than,
-            cmap(m).greater_than,
-            cmap(m).equal_to,
-            cmap(m).interval
-        ]
-        for (key, val) in scalar_affine
-            if val > row
-                scalar_affine[key] -= 1
-            end
-        end
-    end
 
-    for vector_affine in [
-            cmap(m).nonnegatives,
-            cmap(m).nonpositives,
-            cmap(m).zeros,
-            cmap(m).vv_nonnegatives,
-            cmap(m).vv_nonpositives,
-            cmap(m).vv_zeros
-        ]
-        for (key, vals) in vector_affine
-            for (i, val) in enumerate(vals)
-                if val > row
-                    vector_affine[key][i] -= 1
-                end
-            end
+    cmap_m = cmap(m)
+    _shift_references_after_delete_scalar!(cmap_m.less_than, row)
+    _shift_references_after_delete_scalar!(cmap_m.greater_than, row)
+    _shift_references_after_delete_scalar!(cmap_m.equal_to, row)
+    _shift_references_after_delete_scalar!(cmap_m.interval, row)
+
+
+    _shift_references_after_delete_vector!(cmap_m.nonnegatives, row)
+    _shift_references_after_delete_vector!(cmap_m.nonpositives, row)
+    _shift_references_after_delete_vector!(cmap_m.zeros, row)
+    _shift_references_after_delete_vector!(cmap_m.vv_nonnegatives, row)
+    _shift_references_after_delete_vector!(cmap_m.vv_nonpositives, row)
+    _shift_references_after_delete_vector!(cmap_m.vv_zeros, row)
+
+end
+
+# The following two functions d
+function _shift_references_after_delete_scalar!(scalar::Dict, row)
+    vals = scalar.vals
```

Comment:
Let's not use hidden fields. 

Can we write this out explicitly in a for loop without losing performance? It's hard to read what is happening at present.

Reply:
The for loop is about 2X slower than the broadcast!

Reply:
The key is that we can just blindly zip through the vals hidden field.   That is where the majority of the performance is coming from.   There is nothing list ```nonzeros``` in SparseArrays that give you access to the vals without having to use the keys to set.

Reply:
The full benchmark you posted is 2x slower using the for loop!???!

I'll allow it if the function body is as follows (with similar changes to the other instances)
```julia
# The next line loops through the hidden `.vals` field of the `scalar`
# dictionary and  decreases the value by 1 if it is greater than `row`. Using
# the in-place broadcast and looping through the hidden vector is more efficient
# than a naive loop using `values(scalar)`. An equivalent formulation (that is 
# 2x slower) is
#
# for (key, value) in scalar
#     if value > row
#         scalar[key] -= 1
#     end
# end
broadcast!(v -> v > row ? v - 1 : v, scalar.vals, scalar.vals)
```
and we add a test that `cmap(model).xxx` has a field `.vals`, along with a note so that if it breaks we know why.